packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -336,7 +346,9 @@ export class VpcNetwork extends VpcNetworkRef {
 
         // Create an Internet Gateway and attach it if necessary
         if (allowOutbound) {
-            const igw = new cloudformation.InternetGatewayResource(this, 'IGW');
+            const igw = new cloudformation.InternetGatewayResource(this, 'IGW', {
+                tags: this.tags.toToken(true),
```

Comment:
Use a named argument instead to make this more readable. I wonder if a policy-based name for this option such as `subResource: true` instead of a mechanism-based name (`propagateOnly`) will make it clear in what use cases this needs to be used. 

In general, this is a result of a short-cut we are taking in the VPC, where we didn't implement L2s for those sub-resources. Ideally, we should have L2s for `InternetGateway`, `InternetGatewayAttachment`, etc, and they should have a tag manager of their own.

I wonder if the right approach is to create a `TagManager` for each one of those:

```ts
const att = new cloudformation.VPCGatewayAttachmentResource(this, 'VPCGW', {
  // ...
  tags: new TagManager(/* parent */ this).toToken()
});
```

Then, propagation will Just Work™️, I think...